tritium-core/src/test/java/com/palantir/tritium/event/InstrumentationPropertiesTest.java

```diff
@@ -82,4 +105,80 @@ public void invalid() {
+                    return "getSystemPropertySupplier: " + InstrumentationProperties.getSystemPropertySupplier("test")
+                            .asBoolean();
+                },
+                () -> {
+                    latch.countDown();
+                    latch.await();
+                    return "getProperty: " + System.getProperty("instrument.test");
+                },
+                () -> {
+                    latch.countDown();
+                    latch.await();
+                    return "setProperty: " + System.setProperty("instrument.test", "true");
+                },
+                () -> {
+                    latch.countDown();
+                    latch.await();
+                    for (int i = 0; i < 1000; i++) {
+                        System.setProperty("test" + i, "value" + i);
+                    }
+                    return "setProperties: " + System.getProperties();
+                },
+                () -> {
+                    latch.countDown();
+                    latch.await();
+                    InstrumentationProperties.reload();
+                    return "reload";
+                },
+                () -> {
+                    latch.countDown();
+                    latch.await();
+                    return "getSystemPropertySupplier: "
+                            + InstrumentationProperties.getSystemPropertySupplier("test").asBoolean();
+                },
+                () -> {
+                    latch.countDown();
+                    latch.await();
+                    return "isSpecificEnabled: " + InstrumentationProperties.isSpecificEnabled("test");
+                },
+                () -> {
+                    latch.countDown();
+                    latch.await();
+                    return "isGloballyEnabled: " + InstrumentationProperties.isGloballyEnabled();
+                });
+
+        List<ListenableFuture<Object>> futures = tasks.stream()
+                .map(task -> executorService.submit(task))
+                .map(task -> {
+                    Futures.addCallback(task, new FutureCallback<Object>() {
+                        @Override
+                        public void onSuccess(@Nullable Object result) {
+                            System.out.println("result: " + result);
+                        }
+
+                        @Override
+                        public void onFailure(Throwable throwable) {
+                            System.err.println("error: " + throwable);
+                            throwable.printStackTrace(System.err);
+                            assertThat(throwable).isNull();
+                        }
+                    }, MoreExecutors.directExecutor());
+                    return task;
+                })
+                .collect(Collectors.toList());
+
+        await().atMost(Duration.FIVE_SECONDS).untilAsserted(() -> {
+            assertThat(Futures.allAsList(futures).get()).hasSize(futures.size());
```

Comment:
I think we want to assert that these have all succeeded, `allAsList` will include failed futures.